inception/inception-api-annotation-experimental/src/main/java/de/tudarmstadt/ukp/inception/experimental/api/AnnotationSystemAPIImpl.java

```diff
@@ -0,0 +1,418 @@
+    }
+
+    @Override
+    public void handleDocument(ClientMessage aClientMessage) throws IOException {
+        try {
+            CAS cas;
+
+            DocumentMessage message = new DocumentMessage();
+
+            if (aClientMessage.getDocument() == 0L) {
+                // TODO receive random new document
+                cas = getCasForDocument(aClientMessage.getUsername(), aClientMessage.getProject(),
+                        41714);
+                message.setDocument(41714);
+            }
+            else {
+                cas = getCasForDocument(aClientMessage.getUsername(), aClientMessage.getProject(),
+                        aClientMessage.getDocument());
+
+                message.setDocument(
+                        toIntExact(documentService.getSourceDocument(aClientMessage.getProject(),
+                                aClientMessage.getDocument()).getId()));
+            }
+
+            message.setViewportText(getViewportText(aClientMessage, cas));
+
+            message.setAnnotations(
+                    filterAnnotations(getAnnotations(cas, aClientMessage.getProject()),
+                            aClientMessage.getViewport()));
+
+            annotationProcessAPI.handleSendDocumentRequest(message, aClientMessage.getClientName());
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+            createErrorMessage(e.getMessage(), aClientMessage.getClientName());
+        }
+
+    }
+
+    @Override
+    public void handleViewport(ClientMessage aClientMessage) throws IOException
+    {
+        CAS cas = getCasForDocument(aClientMessage.getUsername(), aClientMessage.getProject(),
+                aClientMessage.getDocument());
+
+        ViewportMessage message = new ViewportMessage();
+
+        message.setViewportText(getViewportText(aClientMessage, cas));
+
+        message.setAnnotations(filterAnnotations(getAnnotations(cas, aClientMessage.getProject()),
+                aClientMessage.getViewport()));
+
+        annotationProcessAPI.handleSendViewportRequest(message, aClientMessage.getClientName());
+
+    }
+
+    @Override
+    public void handleSelectAnnotation(ClientMessage aClientMessage) throws IOException
+    {
+        CAS cas = getCasForDocument(aClientMessage.getUsername(), aClientMessage.getProject(),
+                aClientMessage.getDocument());
+        AnnotationFS annotation = selectAnnotationByAddr(cas,
+                aClientMessage.getAnnotationAddress());
+        AnnotationMessage message = new AnnotationMessage(String.valueOf(annotation._id()),
+                annotation.getBegin(), annotation.getEnd(), annotation.getType().getShortName(),
+                annotation.getCoveredText());
+        annotationProcessAPI.handleSendSelectAnnotation(message, aClientMessage.getClientName());
+    }
+
+    @Override
+    public void handleCreateAnnotation(ClientMessage aClientMessage) throws IOException
+    {
+        try {
+            CasStorageSession.open();
```

Comment:
Sessions should be opened in a try-with-resources so that they are guaranteed to be properly closed at the end.